config/v1/types_authentication.go

```diff
@@ -243,11 +244,22 @@ type OIDCProvider struct {
 	// +listType=atomic
 	// +optional
 	ClaimValidationRules []TokenClaimValidationRule `json:"claimValidationRules,omitempty"`
+
+	// userValidationRules defines the set of rules used to validate claims in a user’s token.
+	// These rules determine whether a token subject is considered valid based on its claims.
+	// Each rule is evaluated independently.
+	// See the TokenUserValidationRule type for more information on rule structure.
+	// +listType=atomic
+	// +kubebuilder:validation:MaxItems=64
+	// +optional
+	// +openshift:enable:FeatureGate=ExternalOIDCWithUpstreamParity
+	UserValidationRules []TokenUserValidationRule `json:"userValidationRules,omitempty"`
```

Comment:
What other constraints are there? Looking at the upstream validations, it looks like there is a uniqueness constraint on the CEL expression, we should enforce this as well.